cypcore/Ledger/Graph.cs

```diff
@@ -400,122 +346,97 @@ private BlockGraph CopyBlockGraph(byte[] block, byte[] prevBlock)
         /// 
         /// </summary>
         /// <returns></returns>
-        private Task TrySaveAndPublishBlockGraph(BlockGraph blockGraph)
+        private async Task TryFinalizeBlockGraph(BlockGraph blockGraph)
         {
             Guard.Argument(blockGraph, nameof(blockGraph)).NotNull();
-            var tcs = new TaskCompletionSource<bool>();
-
-            async void TrySave()
+            try
             {
-                try
+                var copy = false;
+                copy |= blockGraph.Block.Node != _serfClient.ClientId;
```

Comment:
Isn't this the same as "var copy = blockGraph.Block.Node != _serfClient.ClientId"?